uportal-war/src/main/resources/properties/db/tables.xml

```diff
@@ -53,6 +53,10 @@
         <name>UPU_DFLT_ID_IDX</name>
         <column-ref>USER_DFLT_USR_ID</column-ref>
     </index>
+    <index>
+        <name>UPU_USER_NAME_IDX</name>
+        <column-ref>USER_NAME</column-ref>
+    </index>
     <unique>
```

Comment:
Yes, this index was needed.

Reply:
If we need it, we need it.  I'm on board ;-).

But I am uncertain about this:  my reading suggests that creating a unique constraint on a column always -- on all the database platforms we might care about -- automatically creates an index on that field.  [Here's an example](https://stackoverflow.com/questions/487314/primary-key-or-unique-index#answer-2916464)

Reply:
This is where I'm confused on this piece of code. In uP4 it is acting as a unique constraint and in uP5 it is acting as a unique index at least for PostgreSQL 9. Hibernate's <unique> seems to be executing differently for us depending on the flavor of uPortal.